### PR TITLE
implemented loop pausing

### PR DESCRIPTION
**Pull request by bcoca**
##### ISSUE TYPE

- Feature Pull Request
##### COMPONENT NAME

loops
##### ANSIBLE VERSION

```
2.2
```
##### SUMMARY

Add a configurable pause to each iteration of a loop.

### CODE REVIEW

File: lib/ansible/executor/task_executor.py [hunk 1]
```diff
@@ -269,6 +270,8 @@ def _run_loop(self, items):
             self._rslt_q.put(TaskResult(self._host.name, self._task._uuid, res), block=False)
             results.append(res)
             del task_vars[loop_var]
+            if loop_pause:
```
Comment (mscherer) [inline lib/ansible/executor/task_executor.py]: Loop_pause would be undefined if self._task.loop_control: is not defined on line 237

File: lib/ansible/executor/task_executor.py [hunk 2]
```diff
@@ -269,6 +270,8 @@ def _run_loop(self, items):
...
+                time.sleep(loop_pause)
```
Comment (mscherer) [inline lib/ansible/executor/task_executor.py]: Also, if there is a error in the task (like a AnsibleParserError), there would be no pause for that iteration

**Comment by mscherer** [comment]
Tests and docs are missing.

File: lib/ansible/executor/task_executor.py (hunk 1 above)
Comment (bcoca) [reply lib/ansible/executor/task_executor.py]: ~~it is inside the if from 237, which ensures that loop_control is defined and has a value~~ missed the other if ... so fixed with declaration above.

File: lib/ansible/executor/task_executor.py (hunk 2 above)
Comment (bcoca) [reply lib/ansible/executor/task_executor.py]: I'm fine with that as the action did not happen

**Pull request merged**